app/serializers/post_serializer.rb

```diff
@@ -223,6 +224,14 @@ def flair_group_id
     object.user&.flair_group_id
   end
 
+  def badges_granted
+    return [] unless SiteSetting.enable_badges && SiteSetting.show_badges_in_post_header
+
+    object.badges_granted.map do |user_badge|
```

Comment:
This will result in the N+1 queries problem since the SQL query will be executed once per post that is being serialized. From what I can tell, the existing pattern seems to be to preload within `TopicView` and access the preloaded data via the `@topic_view` instance variable in the serializer.

Reply:
My bad, I see that you have already eager loaded this in `TopicView`.